nemo/collections/asr/models/asr_model.py

```diff
@@ -136,9 +136,56 @@ def on_after_backward(self):
                 torch.distributed.all_reduce(valid_gradients, op=torch.distributed.ReduceOp.MIN)
 
             if valid_gradients < 1:
-                logging.warning(f'detected inf or nan values in gradients! Setting gradients to zero.')
+                logging.warning('detected inf or nan values in gradients! Setting gradients to zero.')
                 self.zero_grad()
 
+    def on_train_start(self):
```

Comment:
Would you please add some description of this feature in the nemo docs?

Reply:
This is a PyTorch Lightning hook, it's not a part of Nemo. More info here: https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.core.hooks.ModelHooks.html

Reply:
I meant adding some description about this freezing feature and how it works to the nemo docs.

Reply:
Okay, can you please send me a link to where these Nemo docs are?

Reply:
You may add s section in this file which explains how this freezing works:
https://github.com/NVIDIA/NeMo/blob/main/docs/source/asr/configs.rst

Reply:
I have updated the docs as requested